src/pages/swap/helpers/swapConfirm/index.ts

```diff
@@ -0,0 +1,92 @@
+import { StateUpdater } from 'preact/hooks';
+import { ApiPromise } from '@polkadot/api';
+import Big from 'big.js';
+
+import {
+  getInputTokenDetailsOrDefault,
+  InputTokenType,
+  OutputTokenType,
+  OUTPUT_TOKEN_CONFIG,
+} from '../../../../constants/tokenConfig';
+
+import { ExecutionInput } from '../../../../hooks/offramp/useMainProcess';
+import { TokenOutData } from '../../../../hooks/nabla/useTokenAmountOut';
+import { Networks } from '../../../../contexts/network';
+
+import { calculateSwapAmountsWithMargin } from './calculateSwapAmountsWithMargin';
+import { performSwapInitialChecks } from './performSwapInitialChecks';
+import { validateSwapInputs } from './validateSwapInputs';
+
+interface SwapConfirmParams {
+  inputAmountIsStable: boolean;
+  address: `0x${string}` | undefined;
+  fromAmount: Big | undefined;
+  tokenOutAmount: { data: TokenOutData | undefined };
+  api: ApiPromise | null;
+  to: OutputTokenType;
+  from: InputTokenType;
+  selectedNetwork: Networks;
+  fromAmountString: string;
+  setIsInitiating: StateUpdater<boolean>;
+  setInitializeFailed: StateUpdater<boolean>;
+  handleOnSubmit: (executionInput: ExecutionInput) => void;
+  setTermsAccepted: (accepted: boolean) => void;
+}
+
+export function swapConfirm(
+  e: Event,
+  {
+    inputAmountIsStable,
+    address,
+    fromAmount,
+    tokenOutAmount,
+    api,
+    to,
+    from,
+    selectedNetwork,
+    fromAmountString,
+    setIsInitiating,
+    setInitializeFailed,
+    handleOnSubmit,
+    setTermsAccepted,
+  }: SwapConfirmParams,
```

Comment:
I think in this case it would make the code easier to read if we don't already extract the parameters here but rather do
```suggestion
  params: SwapConfirmParams,
```

and then below reference them as `params.address` etc. Only personal preference but I don't like the declaration of the function parameters spanning 16 lines.